--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -707,13 +707,9 @@
   message(STATUS "OpenZWave library found at: ${OpenZWave}")
   target_link_libraries(domoticz ${OpenZWave})

-  find_path(OPENZWAVE_INCLUDE_DIRS NAMES openzwave/Manager.h)
+  find_path(OPENZWAVE_INCLUDE_DIRS NAMES Manager.h)
   if (OPENZWAVE_INCLUDE_DIRS)
-    IF(CMAKE_SYSTEM_NAME STREQUAL "FreeBSD")
-      include_directories(${OPENZWAVE_INCLUDE_DIRS})
-    ELSE()
-      include_directories(${OPENZWAVE_INCLUDE_DIRS}/openzwave)
-    ENDIF()
+    include_directories(${OPENZWAVE_INCLUDE_DIRS})
     message(STATUS "OpenZWave includes found at: ${OPENZWAVE_INCLUDE_DIRS}")
   else()
     message(FATAL_ERROR "OpenZWave includes not found. Did you not issue 'sudo make install' after building OpenZWave?")

