diff -urN domoticz-4.10717.orig/CMakeLists.txt domoticz-4.10717/CMakeLists.txt
--- domoticz-4.10717.orig/CMakeLists.txt	2019-05-10 11:37:55.000000000 +0200
+++ domoticz-4.10717/CMakeLists.txt	2019-10-05 12:18:52.336750288 +0200
@@ -689,7 +689,7 @@
 IF(OpenZWave)
   message(STATUS ${OpenZWave})
   target_link_libraries(domoticz ${OpenZWave})
-  include_directories(${CMAKE_SOURCE_DIR}/hardware/openzwave)
+  include_directories(${OPENZWAVE_INCLUDE_DIRS})
   add_definitions(-DWITH_OPENZWAVE)
   # open-zwave needs libudev if statically linked
   IF(USE_STATIC_OPENZWAVE)
diff -urN domoticz-4.10717.orig/hardware/openzwave/control_panel/ozwcp.cpp domoticz-4.10717/hardware/openzwave/control_panel/ozwcp.cpp
--- domoticz-4.10717.orig/hardware/openzwave/control_panel/ozwcp.cpp	2019-05-10 11:37:55.000000000 +0200
+++ domoticz-4.10717/hardware/openzwave/control_panel/ozwcp.cpp	2019-10-05 12:13:41.340069579 +0200
@@ -39,11 +39,11 @@
 #include <stdlib.h>
 #include <time.h>
 #include <string.h>
-#include "../Options.h"
-#include "../Manager.h"
-#include "../Node.h"
-#include "../Group.h"
-#include "../Notification.h"
+#include <Options.h>
+#include <Manager.h>
+#include <Node.h>
+#include <Group.h>
+#include <Notification.h>
 
 #include <sys/stat.h>
 #include <fstream>
diff -urN domoticz-4.10717.orig/hardware/openzwave/control_panel/ozwcp.h domoticz-4.10717/hardware/openzwave/control_panel/ozwcp.h
--- domoticz-4.10717.orig/hardware/openzwave/control_panel/ozwcp.h	2019-05-10 11:37:55.000000000 +0200
+++ domoticz-4.10717/hardware/openzwave/control_panel/ozwcp.h	2019-10-05 12:14:19.620071306 +0200
@@ -37,8 +37,8 @@
 
 #include <list>
 #include <algorithm>
-#include "Driver.h"
-#include "Notification.h"
+#include <Driver.h>
+#include <Notification.h>
 
 #define MAX_NODES 255
 
