From 3b96ccfcd61ef4e30d4fa246dcd1724c0712bbe9 Mon Sep 17 00:00:00 2001
From: Jan Palus <jpalus@fastmail.com>
Date: Mon, 31 Jan 2022 12:15:41 +0100
Subject: [PATCH] add header for strstr

---
 hardware/EnOceanRawValue.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/hardware/EnOceanRawValue.cpp b/hardware/EnOceanRawValue.cpp
index 1a6769118..14de21632 100644
--- a/hardware/EnOceanRawValue.cpp
+++ b/hardware/EnOceanRawValue.cpp
@@ -1,6 +1,7 @@
 #include "stdafx.h"
 #include "EnOceanRawValue.h"
 #include <string>
+#include <string.h>
 
 
 
-- 
2.35.1

